docs/setup/k8s/running-in-production-k8s.md

```diff
@@ -0,0 +1,108 @@
+# Running Nuclio over Kubernetes in production
+
+After familiarizing yourself with Nuclio, and [deploying it over k8s](/docs/setup/k8s/getting-started-k8s.md) you may find yourself looking for extra configuration knobs, and proper practices for using it in a production environment.
+In [Iguazio](https://www.iguazio.com/) we integrated Nuclio as part of our [Data science PaaS](https://www.iguazio.com/platform/), and it is in extensive use in production, for both our customers and ourselves, running various workloads.
+
+Here you will find more advanced configuration options, and some practices which address the needs of running Nuclio in production environments. 
+
+
+#### In this document
+
+- [Preferred installation method](#preferred-installation-method)
+- [Multi Tenancy](#multi-tenancy)
+- [Dark site operation](#dark-site-operation)
+- [Using Kaniko as an image builder](#using-kaniko-as-an-image-builder)
+
+## Preferred installation method
+
+Even though you may, of course, use any of the available methods, we find that heavy-lifting over k8s is best done with helm charts.
+Nuclio's [helm chart](/hack/k8s/helm/nuclio/) is our preferred mode of deploying Nuclio in [Iguazio](https://www.iguazio.com/) as of writing this document.
+This is also where you'll find most of the production oriented features appearing first, and is the most tightly maintained form of deployment.
+
+Below is a quick example of how to setup the latest stable version of nuclio via helm charts:
+
+- Create a namespace for your functions, and a secret with your intended registry's credentials
+    ```sh
+    kubectl create namespace nuclio
+    ```
+    ```sh
+    read -s mypassword
+    <enter your password>
+    
+    kubectl create secret docker-registry registry-credentials --namespace nuclio \
+        --docker-username <username> \
+        --docker-password $mypassword \
+        --docker-server <URL> \
+        --docker-email ignored@nuclio.io
+    
+    unset mypassword
+    ```
+ - Checkout the nuclio project and install nuclio from its helm chart: 
+    ```sh
+    git checkout https://github.com/nuclio/nuclio.git
+    
+    helm install \
+        --set registry.secretName=registry-credentials \
+        --set registry.pushPullUrl=<your registry URL> \
+        --set controller.image.tag=latest-amd64 \
```

Comment:
You can't use "latest" here. It must be the actual specified version

Reply:
You can, but I would see why we wouldn't, even for an example (please note I'm referring to version freeze under darksite).
Anyways, changed to latest release. Upon further reflection I think version freezing is worth it's own section